sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -467,20 +470,20 @@ class ResolveSessionCatalog(
         replace,
         viewType)
 
-    case ShowTablePropertiesStatement(SessionCatalog(_, tableName), propertyKey) =>
+    case ShowTablePropertiesStatement(SessionCatalogAndTable(_, tableName), propertyKey) =>
       ShowTablePropertiesCommand(
         tableName.asTableIdentifier,
         propertyKey)
 
     case ShowFunctionsStatement(userScope, systemScope, pattern, fun) =>
       val (database, function) = fun match {
-        case Some(CatalogAndIdentifierParts(catalog, functionName)) =>
+        case Some(CatalogObjectIdentifier(catalog, functionIdent)) =>
```

Comment:
Does function identifiers need to handle global_temp, etc.? or just going thru `CatalogObjectIdentifier` is enough?